resharper/resharper-unity/src/Rider/UnityRefreshTracker.cs

```diff
@@ -215,7 +215,7 @@ private void FileSystemChange(FileSystemChange fileSystemChange)
 
         private void AdviseFileAddedOrDeleted(FileSystemChangeDelta delta)
         {
-            if (delta.NewPath.ExtensionNoDot == "cs")
+            if (delta.NewPath.ExtensionNoDot == "cs" || delta.NewPath.ExtensionNoDot == "asmdef")
```

Comment:
btw, what's about .shader, .cginc, .cg and other files which are added to csproj generation at unity side?

Reply:
This is done specifically for the case, when csproj didn't exist and should be created when asmdef or cs file is added. Not sure, if the same is really needed for extensions, which you have mentioned. Still focusing unity or calling Save in Rider would also trigger the same project generation. So it is a matter of convenience for a special case.